src/data/set/image.lean

```diff
@@ -471,6 +471,39 @@ end
 
 end image
 
+/-! ### Lemmas about the powerset and image. -/
+
+section powerset
+
+lemma mem_powerset_insert_iff {s t : set α} {x : α} :
+  s ∈ 𝒫 (insert x t) ↔ s ∈ 𝒫 t ∨ s ∈ insert x '' 𝒫 t :=
+begin
+  simp_rw [mem_powerset_iff],
+  constructor,
+  { intro h,
+    by_cases hs : x ∈ s,
+    { right,
+      use s \ {x},
+      constructor,
+      { rw [mem_powerset_iff, diff_singleton_subset_iff],
+        assumption },
+      { rw [insert_diff_singleton, insert_eq_of_mem hs] }},
+    { left,
+      exact (subset_insert_iff_of_not_mem hs).mp h}},
+  { intro h,
+    rcases h with h | ⟨s', h₁, h₂⟩,
+    { exact subset_trans h (subset_insert x t) },
+    { rw [←h₂],
+      exact insert_subset_insert h₁ }}
+end
+
+/-- The powerset of `{a} ∪ s` is `𝒫 s` together with `{a} ∪ t` for each `t ∈ 𝒫 s`. -/
+theorem powerset_insert (s : set α) (a : α) :
+  𝒫 (insert a s) = 𝒫 s ∪ (𝒫 s).image (insert a) := by
+ext y; rw [mem_powerset_insert_iff, mem_union, mem_image]
```

Comment:
I think you should combine these lemmas into one, the first one is a bit of a weird statement. Also note that:

* `by` at the end of a line is mathlib4 style
* `image` is better spelt `''`

Sorry for taking so long to come back to this PR, please ping me on Zulip next time!

Reply:
No worries, there is no rush. Done.